tests/unit/apps/test_auxiliary.py

```diff
@@ -840,6 +840,37 @@ def test_ovn_workload_ver_lower_than_22_principal(model):
         app.generate_upgrade_plan(target, False)
 
 
+def test_ovn_version_pinning_principal(model):
+    """Test the OvnPrincipal when enable-version-pinning is set to True."""
+    target = OpenStackRelease("victoria")
+    charm = "ovn-central"
+    exp_msg = f"'{charm}' should set 'enable-version-pinning' before upgrading"
+    machines = {"0": MagicMock(spec_set=COUMachine)}
+    app = OvnPrincipal(
+        name=charm,
+        can_upgrade_to="22.03/stable",
+        charm=charm,
+        channel="22.03/stable",
+        config={"source": {"value": "distro"}, "enable-version-pinning": {"value": True}},
+        machines=machines,
+        model=model,
+        origin="ch",
+        series="focal",
+        subordinate_to=[],
+        units={
+            f"{charm}/0": COUUnit(
+                name=f"{charm}/0",
+                workload_version="22.03.2",
+                machine=machines["0"],
+            )
+        },
+        workload_version="22.03.2",
+    )
+
+    with pytest.raises(ApplicationError, match=exp_msg):
+        app.generate_upgrade_plan(target, False)
```

Comment:
I do not see the point of running the whole `generate_upgrade_plan` and not simply `upgrade_plan_sanity_checks`.